src/shogun/converter/ica/FastICA.cpp

```diff
@@ -76,6 +76,7 @@ CFeatures* CFastICA::apply(CFeatures* features)
 	SG_REF(features);
 
 	SGMatrix<float64_t> X = ((CDenseFeatures<float64_t>*)features)->get_feature_matrix();
+	REQUIRE(X.data(), "you are using an empty matrix");
```

Comment:
it's good but the error message should be rather: "Features have not been provided.\n"